docs/framework_events.rst

```diff
@@ -4,6 +4,29 @@
 Custom Events
 =============
 
+Bank
+^^^^
+
+.. py:method:: red_bank_set_global(global_state: bool)
+
+    Dispatched when the Bank gets toggled between Global and Local. :code:`global_state` will be True if the Bank is being set to Global or False if the bank is being set to Local
+
+.. py:method:: red_bank_set_balance(payload: redbot.core.bank.BankSetBalanceInformation)
+
+    Dispatched when a user has their balance changed.
+
+.. py:method:: red_bank_transfer_credits(payload: redbot.core.bank.BankTransferInformation)
+
+    Dispatched when credits gets transfered from one user to another.
+
+.. py:method:: red_bank_wipe(guild_id: int)
+    
+    Dispatched when the Bank gets wiped. :code:`guild_id` will be the ID of the Guild that was wiped, 0 if all Guilds were wiped (Bank is Local), or -1 if all Users were wiped (Bank is Global)
```

Comment:
Having both 0 and -1 as special values seems a bit smelly. Is there a meaningful difference for the CC to account for that makes differentiating between wiping guilds or users something that needs to happen? If so, I think it would be neater to just have a second argument, whether or not the bank was global at the time of wiping.

I would say the CC can just check if the bank is global or not to get the same value, but that only applies for `[p]bankset reset`. `[p]bankset toggleglobal` also implicitly results in this being called, since it clears the bank during the process of swapping scope. Trying to test for it at that time would likely result in an incorrect value due to race conditions.